src/domain/Lists/DraggableList/DraggableList.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react'
+import {
+  DragDropContext,
+  Draggable,
+  DropResult,
+  Droppable
+} from 'react-beautiful-dnd'
+import uuid from 'uuid'
+
+import { Props } from '../../../common'
+import { StyledListArea, StyledListItem } from './style'
+
+interface IDraggableListProps<T> {
+  /** The array of items */
+  items: T[]
+  /** Function to update the array of items when they have been re-arranged */
+  setItems: (nodes: T[]) => void
+  /** An array of ReactNode that should represent the array of items */
+  children: React.ReactNode[]
+  /** The data-component-context */
+  componentContext?: string
+}
+
+const DraggableList = <T extends {}> ({ items, setItems, children, componentContext }: IDraggableListProps<T>) => {
+  const onDragEnd = (result: DropResult) => {
+    if (!result.destination) {
+      return
+    }
+    const { source, destination } = result
+
+    const newItems = [...items]
+    const [removed] = newItems.splice(source.index, 1)
+    newItems.splice(destination.index, 0, removed)
+    setItems(newItems)
+  }
+
+  return (
+    <span
+      data-component-type={Props.ComponentType.DraggableList}
+      data-component-context={componentContext}
+    >
+      <DragDropContext onDragEnd={onDragEnd}>
+        <Droppable droppableId={uuid.v4()}>
+          {(droppableProvided) => (
+            <div ref={droppableProvided.innerRef}>
+              <StyledListArea>
+                {children.map((child, index) => (
+                  <Draggable
+                    key={index}
```

Comment:
I worry about using index as a key maybe uuid would be great. I am not sure but when deleting is implemented in the future, it might not work

Reply:
for example, if keys are 0,1,2,3. and we delete the third item, then it will leave items with key of 0,1,3 but new index is 0,1,2.